### PR TITLE
Exposing prefetchCount through clients

### PR DESCRIPTION
**Pull request by nemakam**
## Description
Exposing PrefetchCount through clients.
Added tests for prefetch.
Fix for #135 

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: test/Microsoft.Azure.ServiceBus.UnitTests/SubscriptionClientTests.cs [hunk 1]
```diff
@@ -194,5 +194,39 @@ public sealed class SubscriptionClientTests : SenderReceiverClientTestBase
...
+                subscriptionClient.PrefetchCount = 3;
+                Assert.Equal(3, subscriptionClient.PrefetchCount);
+                Assert.Equal(3, subscriptionClient.ServiceBusConnection.PrefetchCount);
+
+                // Already created message receiver should have its prefetch value updated.
+                Assert.Equal(3, subscriptionClient.InnerSubscriptionClient.InnerReceiver.PrefetchCount);
+            }
+            finally
+            {
+                subscriptionClient.CloseAsync().Wait();
```
Comment (jtaubensee) [inline test/Microsoft.Azure.ServiceBus.UnitTests/SubscriptionClientTests.cs]: Any reason not to await?

Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/SubscriptionClientTests.cs]: @jtaubensee why `Wait()`? Change method to return a `Task` and use proper await

```c#
await subscriptionClient.CloseAsync();
```

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs [hunk 2]
```diff
@@ -56,6 +56,23 @@ public MessageReceiver InnerReceiver
             }
         }
 
+        /// <summary>
+        /// Gets or sets the number of messages that the queue client can simultaneously request.
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: >queue client [](start = 57, length = 12)

nit: subscription client

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs [hunk 3]
```diff
@@ -56,6 +56,23 @@ public MessageReceiver InnerReceiver
...
+        /// </summary>
+        /// <value>The number of messages that the queue client can simultaneously request.</value>
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: >queue client [](start = 51, length = 12)

same here

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs [hunk 4]
```diff
@@ -56,6 +56,23 @@ public MessageReceiver InnerReceiver
...
+        public int PrefetchCount
+        {
+            get => this.ServiceBusConnection.PrefetchCount;
```
Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs]: >get => this.ServiceBusConnection.PrefetchCount; [](start = 12, length = 47)

nit: add a line b/n get and set

**Comment by vinaysurya** [comment]
:shipit:

**Pull request merged**